ui/app/adapters/allocation.js

```diff
@@ -6,12 +6,17 @@ import classic from 'ember-classic-decorator';
 export default class AllocationAdapter extends Watchable {
   stop = adapterAction('/stop');
 
-  restart(allocation, taskName) {
+  restart(allocation, taskName, allTasks) {
     const prefix = `${this.host || '/'}${this.urlPrefix()}`;
     const url = `${prefix}/client/allocation/${allocation.id}/restart`;
-    return this.ajax(url, 'PUT', {
-      data: taskName && { TaskName: taskName },
-    });
+    let data = {};
+    if (taskName) {
+      data.TaskName = taskName;
+    }
+    if (allTasks) {
+      data.AllTasks = !!allTasks;
```

Comment:
`question`: Why is this a boolean?

Reply:
This API expects a boolean: https://github.com/hashicorp/nomad/pull/14127/files#diff-b4fa614bde4cb3604e2b5a1b95d00e4381a06e6e1cb1a29f2c34fe44a0444d01R468

Is there a better way to make sure a param is a boolean? That's how I used to do it 😅